janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java

```diff
@@ -356,17 +358,23 @@ public void deleteIndex(String indexName) throws IOException {
     public void bulkRequest(List<ElasticSearchMutation> requests, String ingestPipeline) throws IOException {
         final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
         for (final ElasticSearchMutation request : requests) {
-            final Map<String, String> requestData;
-            if(useMappingTypes){
-                requestData = ImmutableMap.of(
+            Map<String, String> requestData = new HashMap<>();
+            if (useMappingTypes) {
+                requestData.putAll(ImmutableMap.of(
                     "_index", request.getIndex(),
                     "_type", request.getType(),
-                    "_id", request.getId());
+                    "_id", request.getId()));
             } else {
-                requestData = ImmutableMap.of(
+                requestData.putAll(ImmutableMap.of(
                     "_index", request.getIndex(),
-                    "_id", request.getId());
+                    "_id", request.getId()));
+            }
+
+            if (request.getRequestType() == ElasticSearchMutation.RequestType.UPDATE) {
+                final String retryOnConflictKey = majorVersion.getValue() >= 7 ? "retry_on_conflict" : "_retry_on_conflict";
```

Comment:
Also, I suggest to make `retryOnConflictKey` as a global variable and assign value to it in the constructor if there is a different `retryOnConflictKey` between different ES versions.

Reply:
Agreed, changes done accordingly.